### PR TITLE
HEEDLS-768 Add convert from Pascal case for resource types

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-768

### Description
Added a convert from PascalCase to space separated method in DisplayStringHelper to use where we display the ResourceTypes from the LearningHub API after the change in https://softwiretech.atlassian.net/browse/HEEDLS-766.

Since it was a small change, I've just updated the HEE code where these was being used in the Frameworks as well, but I'm not 100% sure that I've got them all. @kevwhitt-hee it's probably worth looking over this to see if I've missed any. I can remove the changes in the Frameworks area from this PR if you'd rather handle these changes from the HEE side.

### Screenshots
image
image
image
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/appsettings.Development.json [hunk 1]
```diff
@@ -17,5 +17,5 @@
     "PricingPage": true,
     "RefactoredFindYourCentrePage": true
   },
-  "LearningHubOpenAPIBaseUrl": "https://uks-learninghubnhsuk-openapi-test.azurewebsites.net"
+  "LearningHubOpenAPIBaseUrl": "https://uks-learninghubnhsuk-openapi-dev.azurewebsites.net"
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Web/appsettings.Development.json:20]: This was intentionally pointing to Test - Test is in sync with the Findwise collection (more or less), whereas Dev isn't, so the search endpoint gives results which are much more consistent with Prod if we point to Test.

Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/appsettings.Development.json:20]: Ah whoops, I didn't mean to commit this change

File: DigitalLearningSolutions.Web/ViewModels/LearningPortal/Completed/CompletedLearningResourceViewModel.cs [hunk 2]
```diff
@@ -9,16 +10,17 @@ public CompletedLearningResourceViewModel(CompletedActionPlanResource resource)
...
             ResourceDescription = resource.ResourceDescription;
             CatalogueName = resource.CatalogueName;
-            ResourceType = resource.ResourceType;
+            ResourceType = DisplayStringHelper.AddSpacesToPascalCaseString(resource.ResourceType);
             ResourceReferenceId = resource.ResourceReferenceId;
             AbsentInLearningHub = resource.AbsentInLearningHub;
         }
 
+        public int ResourceReferenceId { get; set; }
+
```
Comment (GeorgeBlackwell) [inline DigitalLearningSolutions.Web/ViewModels/LearningPortal/Completed/CompletedLearningResourceViewModel.cs:19]: We should match the formatting here with the rest of the properties

File: DigitalLearningSolutions.Web/ViewModels/LearningPortal/Current/CurrentLearningResourceViewModel.cs [hunk 3]
```diff
@@ -2,24 +2,29 @@
...
             ResourceDescription = resource.ResourceDescription;
             CatalogueName = resource.CatalogueName;
-            ResourceType = resource.ResourceType;
+            ResourceType = DisplayStringHelper.AddSpacesToPascalCaseString(resource.ResourceType);
             AbsentInLearningHub = resource.AbsentInLearningHub;
             ResourceReferenceId = resource.ResourceReferenceId;
         }
 
+        public int ResourceReferenceId { get; set; }
+
```
Comment (GeorgeBlackwell) [inline DigitalLearningSolutions.Web/ViewModels/LearningPortal/Current/CurrentLearningResourceViewModel.cs:23]: Same here

**Review by GeorgeBlackwell — COMMENTED** [review]
LGTM, just a few formatting suggestions

File: DigitalLearningSolutions.Web.Tests/Helpers/DisplayStringHelperTests.cs [hunk 4]
```diff
@@ -310,5 +310,21 @@ public void GetTimeStringFromMinutes_returns_expected_result(int minutes, string
...
             result.Should().Be(expectedResult);
         }
+
+        [Test]
+        [TestCase("", "")]
+        [TestCase("NicePascalCaseString", "Nice Pascal Case String")]
+        [TestCase("ASlightlyMoreAwkwardOne", "A Slightly More Awkward One")]
+        [TestCase("XMLFile", "XML File")]
+        [TestCase("An Already Separated String", "An Already Separated String")]
+        [TestCase("A Partially SeparatedString", "A Partially Separated String")]
```
Comment (JBsoftwire) [inline DigitalLearningSolutions.Web.Tests/Helpers/DisplayStringHelperTests.cs:320]: Does it work for strings starting with 'ID' or which have 'XML' not at the start? (if not, how exhaustive do we want to get? are we even worried about those cases?)

**Review by JBsoftwire — CHANGES_REQUESTED** [review]
Just a question on top of what George said.

**Pull request merged**